src/games/ecards.py

```diff
@@ -0,0 +1,275 @@
+import asyncio
+from abstracts import EGameFactory
+
+import random
+
+from interactive import *
+
+from utils import dmerge
+
+
+class ECards(EGameFactory):
+    """
+    EGameFactory for a game where one player is given a prompt,
+    the other players select an answer from a set of safety answers,
+    and the head player selects their favourite answer.
+    """
+
+    # Configuration for the game
+    game_name = "E Cards"
+    game_description = "Prompt cards with your friends!"
+    wait_duration = 6
+    min_players = 2
+    cog_help = """
+    EGameFactory for a game where one player is given a prompt,
+    the other players select an answer from a set of safety answers,
+    and the head player selects their favourite answer.
+    """
+    has_scrape = True
+
+    file_prompts = "data/elash_prompts_{gid}.txt"
+    file_safeties = "data/elash_safeties_{gid}.txt"
+
+    def __init__(self, context):
+        super().__init__(context, __name__)
+
+        # instance property of all prompts
+        self.prompts = None
+        # instance property of all safety "cards"
+        self.safeties = None
+
+    async def start(self):
+        """
+        Method for starting the E Card game.
+        Reads in prompts, puts the deck in a particular state, and runs the game.
+        """
+
+        # check if prompts
+        if self.prompts == self.safeties == None:
+            # info message
+            self.logging.info(f"Reading prompts and safety files for {self.guild.id}.")
+            await self.channel.send(
+                embed=self.embed(f"Reading prompts and safety files...")
+            )
+
+            # read files
+            try:
+                self.prompts = self._read_file(self.file_prompts)
+                self.safeties = self._read_file(self.file_safeties)
+            except Exception as e:
+                self.logging.error(f"Error reading files {e}")
+                # early exit
+                return await self.channel.send(
+                    embed=self.embed(f"Error reading files\n```\n{e}\n```")
+                )
+
+            self.logging.info(
+                f"Read in {len(self.prompts)} prompts and {len(self.safeties)} safeties."
+            )
+
+        # create deck orders
+        prompt_deck = random.sample(self.prompts, len(self.prompts))
+        answer_deck = random.sample(self.safeties, len(self.safeties))
+
+        # create starting hands
+        hands = {pid: [answer_deck.pop() for _ in range(5)] for pid in self.players}
```

Comment:
It's a flavour thing, but when iterating over dictionaries, please use one of the accessor methods (i.e. `.keys()`). Sure it's a couple more lines of bytecode, but the readibility, I think at least, is worth it. :)